### PR TITLE
chore(core): move common/core/desktop/ to core/ 🍏

### PR DESCRIPTION
**Pull request by mcdurdin**
Moves common/core/desktop/ to core/ and updates scripts and paths.

This is based currently on the 🎡 PR chain but am starting a new chain as it is no longer web-related.

This will not be merged until 🎡 is approved and merged.

# User Testing

This reorganisation chain impacts Core, Windows and Developer. This means we need to run regression tests on the impacted products. I do not anticipate serious issues but there is potential for errors that are not picked up in the build process. Developer tests will be run on the 🎢 chain.

* **TEST_WINDOWS**: Please run a full regression test on Keyman for Windows.
* **TEST_LINUX**:  Please run a full regression test on Keyman for Linux.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -18,16 +18,19 @@ android/samples/: android/Samples/**
...
-common/core/: common/core/**
-common/core/desktop/: common/core/desktop/**
 
 common/models/: common/models/**
 common/models/types/: common/models/types/**
 common/models/templates/: common/models/templates/**
 common/models/wordbreakers/: common/models/wordbreakers/**
 
 common/resources/: resources/**
+common/web/: core/web/**
```
Comment (mcdurdin) [inline .github/labeler.yml:30]:
```suggestion
common/web/: common/web/**
```

**Review by jahorton — COMMENTED** [review]
My first-pass review:  base changeset seems fine.

**Review by ermshiperete — APPROVED** [review]

File: resources/build/trigger-definitions.inc.sh [hunk 2]
```diff
@@ -11,12 +11,12 @@ available_platforms=(android ios linux mac web windows)
 
 watch_android='web|common/models|common/predictive-text|common/lexical-model-types|common/web'
 watch_ios='web|common/models|common/predictive-text|common/lexical-model-types|common/web'
-watch_linux='common/core/desktop'
-watch_mac='common/core/desktop'
+watch_linux='core'
+watch_mac='core'
 watch_web='common/models|common/predictive-text|common/lexical-model-types|common/web'
 
 # Windows currently builds Developer and Desktop, so we need everything from common,developer,web
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply resources/build/trigger-definitions.inc.sh:18]: I'll do this on the 🎢 chain :grin: which is where the Developer build split takes effect.

**Comment by mcdurdin** [comment]
# Test Results

* **TEST_WINDOWS (OPEN):** See #6764
* **TEST_LINUX (OPEN):** See #6765

**Comment by bharanidharanj** [comment]
@mcdurdin The Build for Windows does not exist in the given Artifacts. Could you please, attach the PR build for testing? Thanks.

**Comment by mcdurdin** [comment]
> @mcdurdin The Build for Windows does not exist in the given Artifacts. Could you please, attach the PR build for testing? Thanks.

This should be coming soon as there are new builds running right now.

**Comment by mcdurdin** [comment]
> > @mcdurdin The Build for Windows does not exist in the given Artifacts. Could you please, attach the PR build for testing? Thanks.
> 
> This should be coming soon as there are new builds running right now.

PR builds for all platforms should now be available @bharanidharanj

**Comment by bharanidharanj** [comment]
> 

@mcdurdin Okay, Thanks.

**Comment by bharanidharanj** [comment]
* **TEST_WINDOWS (PASSED):** Tested with 16.0.13-alpha-test-6629 build in Windows 10 OS and it is working as expected.

**Comment by mcdurdin** [comment]
* **TEST_LINUX (PASSED):** While #6765 uncovered some issues in Linux, none of them are arising from this PR. We'll mark this as PASSED here and track those issues separately.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.18-alpha](https://keyman.com/downloads/releases/alpha/16.0.18)

**Pull request merged**